vmware/container-service-extension
# [VCDA-2193] Change CSE install and upgrade workflow to install supported templates

To help us process your pull request efficiently, please include: 

* Cache only the templates which are supported by the current CSE version if legacy_mode is false
* install and upgrade should only install templates supported by the currently running CSE.
* Template is supported if template['min_cse_version'] >= current_cse_version <= template['max_cse_version']
* Store template's min cse version and max cse version as part of template metadata

Testing done:
* Check if CSE install installs only the supported templates
* Check if correct error is thrown if user tries to install a template which is not supported.

Required:
* Should be merged after in-place tag 3.1 is created.

@rocknes @sahithi @sakthisunda @arunmk


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/container-service-extension/958)
